### PR TITLE
PIT-339: Change PR template language around testing requirements

### PR DESCRIPTION
**Pull request by tariksbl**
Why this PR is needed
----
Change the language of the pull request template from "How to test / verify these changes" to "How this change was tested", and replace "optional!" with "..or why testing isn't applicable" .

Jira ticket reference : [PIT-339](https://energyhub.atlassian.net/browse/PIT-339)

What this PR includes
----
Brief description of what was done to meet the PR goal.

How this change was tested
----
I couldn't test via creating a branch off of this branch (and [docs](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/about-issue-and-pull-request-templates#pull-request-templates) suggest the same).  But- the change is small.

[PIT-339]: https://energyhub.atlassian.net/browse/PIT-339?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: pull_request_template.md [hunk 1]
```diff
@@ -8,6 +8,6 @@ What this PR includes
 ----
 Brief description of what was done to meet the PR goal.
 
-How to test / verify these changes
+How this change was tested
 ----
-This is an optional section! Only include it if applicable.
\ No newline at end of file
+Describe testing done on the change, or why this is not applicable.
```
Comment (pcalouche) [inline pull_request_template.md:13]: How is "How this change was tested" different from "Describe testing done on the change"?

Comment (tariksbl) [reply pull_request_template.md:13]: Section header, followed by instructions for that section.  Is this what you mean?

Comment (pcalouche) [reply pull_request_template.md:13]: Ignore my comment.  I looked at the diff completely wrong. I thought it was adding a new section.

**Review by ryankubin — APPROVED** [review]
LGTM, consensus agreed within eng.

**Pull request merged**